addons/edi_sale/models/edi_sale_line_request_record.py

```diff
@@ -74,6 +74,11 @@ def sale_line_values(self):
 
     def execute(self):
         """Execute records"""
+        # Some subclasses also inherit 'edi.record.sync' and want to create/update via
+        # synchronizer. Don't create records in such cases
+        if self.env.context.get('update_via_sync', False):
```

Comment:
```suggestion
        if self._context.get("update_via_sync"):
```

Reply:
Shouldn't that default value `.get(..., False)` be in place? This method may get called from other subclasses which do not set the value.

Reply:
The default default value is `None`, so you should still get a false-y result if the key is missing. Personally I don't mind an explicit `False` here - it's a strong hint that we expect a boolean value if the key exists - but it isn't strictly necessary.

Reply:
Yes as Kevin's says it will be None if key is not in context which still will work, unless we explicitly check if == False

Reply:
Updated, please review.